src/backend/libs/efiling-account-client/src/test/java/ca/bc/gov/open/jag/efilingaccountclient/csoStatusServiceImpl/GetDocumentDetailsTest.java

```diff
@@ -1,29 +1,105 @@
+import ca.bc.gov.ag.csows.filing.status.NestedEjbException_Exception;
+import ca.bc.gov.open.jag.efilingaccountclient.CSODocumentServiceImpl;
+import ca.bc.gov.open.jag.efilingcommons.exceptions.EfilingDocumentServiceException;
+import ca.bc.gov.open.jag.efilingcommons.model.DocumentDetails;
 import org.junit.jupiter.api.*;
 import org.mockito.Mock;
+import org.mockito.Mockito;
 import org.mockito.MockitoAnnotations;
 
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+import static org.mockito.ArgumentMatchers.any;
+
 @TestInstance(TestInstance.Lifecycle.PER_CLASS)
 @DisplayName("Get Document Details Test Suite")
 public class GetDocumentDetailsTest {
 
+    private static final String DOCUMENT_TYPE_CD = "ACODE";
+    private static final String DESCRIPTION = "DESCRIPTION";
+    private static final String NODOC = "NODOC";
+    public static final String COURT_LEVEL = "level1";
+    public static final String COURT_CLASS = "class1";
+    public static final String EXCEPTION = "exception";
+    @Mock
+    FilingStatusFacadeBean filingStatusFacadeBean;
 
-    private static CSOStatusServiceImpl sut;
+    private static CSODocumentServiceImpl sut;
 
     @BeforeAll
-    public void setUp() {
+    public void setUp() throws NestedEjbException_Exception {
 
         MockitoAnnotations.initMocks(this);
+        DocumentType documentType = new DocumentType();
+        documentType.setDocumentTypeCd(DOCUMENT_TYPE_CD);
+        documentType.setDocumentTypeDesc(DESCRIPTION);
+        documentType.setDefaultStatutoryFee(BigDecimal.TEN);
+
+        Mockito.when(filingStatusFacadeBean.getDocumentTypes(Mockito.eq(COURT_LEVEL),any())).thenReturn(Arrays.asList(documentType));
+        Mockito.when(filingStatusFacadeBean.getDocumentTypes(Mockito.eq(NODOC),any())).thenReturn(new ArrayList<>());
+
+        Mockito.when(filingStatusFacadeBean.getDocumentTypes(Mockito.eq(EXCEPTION),any())).thenThrow(NestedEjbException_Exception.class);
 
-        sut = new CSOStatusServiceImpl();
+        sut = new CSODocumentServiceImpl(filingStatusFacadeBean);
     }
 
-    @DisplayName("OK: test returns null ")
+    @DisplayName("OK: test returns document ")
     @Test
-    public void testWithFileRoleEnabled() {
-        Assertions.assertNull(sut.getDocumentDetails(""));
+    public void testWithFoundResult() {
+        DocumentDetails result = sut.getDocumentDetails(COURT_LEVEL, COURT_CLASS, DOCUMENT_TYPE_CD);
+        Assertions.assertEquals(DESCRIPTION, result.getDescription());
+        Assertions.assertEquals(BigDecimal.TEN, result.getStatutoryFeeAmount());
+    }
+
+    @DisplayName("Exception: when not finding document should throw exception")
+    @Test
+    public void whenNotFindingDocumentsShouldThrowError() {
+
+        Assertions.assertThrows(EfilingDocumentServiceException.class, () -> sut.getDocumentDetails(NODOC, COURT_CLASS, NODOC));
     }
+
+    @DisplayName("Failure: throws exception")
```

Comment:
question: can we specify what kind of failure here in the display name?

Reply:
done